### PR TITLE
Simplify the JSON and protobuf protocols

### PR DESCRIPTION
**Pull request by davidfowl**
- Remove message type and message format
- Updated tests

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/TestConnection.cs [hunk 1]
```diff
@@ -79,11 +81,18 @@ public Task StartAsync(ITransportFactory transportFactory, HttpClient httpClient
...
             var json = JsonConvert.SerializeObject(jsonObject, Formatting.None);
-            var bytes = Encoding.UTF8.GetBytes($"{json.Length}:T:{json};");
+            var bytes = FormatMessageToArray(Encoding.UTF8.GetBytes(json));
 
             return _receivedMessages.Out.WriteAsync(bytes);
         }
 
+        private byte[] FormatMessageToArray(byte[] message, int bufferSize = 1024)
+        {
+            var output = new ArrayOutput(1024);
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/TestConnection.cs]: use bufferSize

File: test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs [hunk 2]
```diff
@@ -134,10 +136,16 @@ public void InvalidMessagesWithBinder(string input, string expectedMessage)
...
         {
-            input = $"{input.Length}:T:{input};";
-            return input;
+            var data = Encoding.UTF8.GetBytes(input);
+            return Encoding.UTF8.GetString(FormatMessageToArray(data));
         }
 
+        private static byte[] FormatMessageToArray(byte[] message, int bufferSize = 1024)
+        {
+            var output = new ArrayOutput(1024);
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs]: use bufferSize

File: src/Microsoft.AspNetCore.Sockets.Http/Transports/LongPollingTransport.cs [hunk 3]
```diff
@@ -35,11 +35,7 @@ public LongPollingTransport(ReadableChannel<byte[]> application, ILoggerFactory
                     return;
                 }
 
-                var headers = context.Request.GetTypedHeaders();
-                var messageFormat = headers.Accept?.Contains(new Net.Http.Headers.MediaTypeHeaderValue(ContentTypes.BinaryContentType)) == true ?
-                MessageFormat.Binary :
-                MessageFormat.Text;
-                context.Response.ContentType = ContentTypes.GetContentType(messageFormat);
+                // REVIEW: What should the content type be?
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/Transports/LongPollingTransport.cs]: Strawman:

* `application/vnd.microsoft.signalr.v1+json`
* `application/vnd.microsoft.signalr.v1+protobuf`
* `application/vnd.microsoft.signalr.v1+msgpack`

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Http/Transports/LongPollingTransport.cs]: Can't specify that here. The transport doesn't know.

Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/Transports/LongPollingTransport.cs]: Then remove the comment?

Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/Transports/LongPollingTransport.cs]: Never mind, I see what you mean.

**Pull request merged**